schematic/__init__.py

```diff
@@ -1,2 +1,114 @@
+import logging
+import os
+import time
+from typing import Dict, List
+
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.instrumentation.flask import FlaskInstrumentor
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import (
+    BatchSpanProcessor,
+    ConsoleSpanExporter,
+    SimpleSpanProcessor,
+    Span,
+)
+from opentelemetry.sdk.trace.sampling import ALWAYS_OFF
+from synapseclient import Synapse
+from werkzeug import Request
+
 from schematic.configuration.configuration import CONFIG
 from schematic.loader import LOADER
+from schematic_api.api.security_controller import info_from_bearer_auth
+
+Synapse.allow_client_caching(False)
+logger = logging.getLogger(__name__)
+
+
+# borrowed from: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/tests/integration/conftest.py
+class FileSpanExporter(ConsoleSpanExporter):
+    """Create an exporter for OTEL data to a file."""
+
+    def __init__(self, file_path: str) -> None:
+        """Init with a path."""
+        self.file_path = file_path
+
+    def export(self, spans: List[Span]) -> None:
+        """Export the spans to the file."""
+        with open(self.file_path, "a", encoding="utf-8") as f:
+            for span in spans:
+                span_json_one_line = span.to_json().replace("\n", "") + "\n"
+                f.write(span_json_one_line)
+
+
+def set_up_tracing() -> None:
+    """Set up tracing for the API."""
+    tracing_export = os.environ.get("TRACING_EXPORT_FORMAT", None)
+    if tracing_export is not None and tracing_export:
+        Synapse.enable_open_telemetry(True)
+        tracing_service_name = os.environ.get("TRACING_SERVICE_NAME", "schematic-api")
+
+        trace.set_tracer_provider(
+            TracerProvider(
+                resource=Resource(attributes={SERVICE_NAME: tracing_service_name})
+            )
+        )
+        FlaskInstrumentor().instrument(
+            request_hook=request_hook, response_hook=response_hook
+        )
```

Comment:
Most all the code in this script was just moved except for this `FlaskInstruement` and the request/response hook. This is coming from this library: https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-flask - Which allows auto instrumentation of the flask server that is running here